settings/provisioner/foreman/foreman.spec

```diff
@@ -0,0 +1,75 @@
+---
+subparsers:
+    foreman:
+        help: Provision systems using 'foreman'
+        groups:
+            - title: foreman server
+              options:
+                  username:
+                      type: str
+                      help: User to Forman server
+                      default: rhos-qe
+                  password:
+                      type: str
+                      help: Password of Forman server
+                      required: True
+                  auth-url:
+                      type: str
+                      help: The Foreman api url
+                      required: True
+                  mgmt-strategy:
+                      type: str
+                      help: Whether to use Foreman or system ipmi command.
+                      default: foreman
+                  mgmt-action:
+                      type: str
+                      help: Which command to send with the power-management selected by mgmt_strategy. For example - reset, reboot, cycle
+                      default: cycle
+                  wait-for-host:
+                      action: store_true
+                      help: Whether we should wait for the host given the 'rebuild' state was set.
+            - title: host
```

Comment:
The current implementation supports only 1 host,
I've opened a ticket for enhancements : https://projects.engineering.redhat.com/browse/RHOSINFRA-83